go.mod

```diff
@@ -1,26 +1,97 @@
 module github.com/vmware/cluster-api-provider-cloud-director
 
-go 1.16
+go 1.17
 
 require (
 	github.com/antihax/optional v1.0.0
 	github.com/apparentlymart/go-cidr v1.1.0
-	github.com/go-openapi/errors v0.20.0
+	github.com/go-openapi/errors v0.20.2
 	github.com/google/uuid v1.3.0
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.15.0
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.19.0
 	github.com/peterhellberg/link v1.1.0
 	github.com/pkg/errors v0.9.1
-	github.com/replicatedhq/troubleshoot v0.25.0
-	github.com/stretchr/testify v1.7.0
-	github.com/vmware/go-vcloud-director/v2 v2.14.0-rc.3
-	golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
+	github.com/replicatedhq/troubleshoot v0.31.4
+	github.com/stretchr/testify v1.7.1
+	github.com/vmware/go-vcloud-director/v2 v2.15.0
+	golang.org/x/oauth2 v0.0.0-20220411215720-9780585627b5
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.22.4
-	k8s.io/apimachinery v0.22.4
-	k8s.io/client-go v0.22.4
+	k8s.io/api v0.23.6
+	k8s.io/apimachinery v0.23.6
+	k8s.io/client-go v0.23.6
 	k8s.io/klog v1.0.0
-	k8s.io/klog/v2 v2.9.0
-	sigs.k8s.io/cluster-api v0.4.2
-	sigs.k8s.io/controller-runtime v0.10.3
+	k8s.io/klog/v2 v2.60.1
+	sigs.k8s.io/cluster-api v1.1.3
+	sigs.k8s.io/controller-runtime v0.11.2
+)
+
+require (
```

Comment:
Is there any difference between this and the top require block?

Reply:
This was added by the `go mod --compat=1.17`. I have not dug much into it.